<?xml version="1.0"?>
<launch>
    <node pkg="challenge_2" exec="dc_motor" name="motor_sys" namespace="gpo1">
      <param name="sample_time" value="0.01"/>
      <param name="K" value="2.16"/>
      <param name="tau" value="0.05"/>
      <param name="initial_conditions" value="0.0"/>
    </node>
    <node pkg="challenge_2" exec="controller" name="ctrl" namespace="gpo1">
      <param name="Kp" value="0.02255"/>
      <param name="Ki" value="1.1563"/>
      <param name="Kd" value="0.00005625"/>
    </node>

    <node pkg="challenge_2" exec="set_point" name="sp_gen" namespace="gpo1">
      <param name="amplitud" value="1.0"/>
      <param name="omega" value="1.0"/>
      <param name="timer_period" value="0.1"/>
      <param name="signal_type" value="1"/>
    </node>

    <node pkg="challenge_2" exec="dc_motor" name="motor_sys" namespace="gpo2">
      <param name="sample_time" value="0.01"/>
      <param name="K" value="2.16"/>
      <param name="tau" value="0.05"/>
      <param name="initial_conditions" value="0.0"/>
    </node>
    <node pkg="challenge_2" exec="controller" name="ctrl" namespace="gpo2">
      <param name="Kp" value="0.02255"/>
      <param name="Ki" value="1.1563"/>
      <param name="Kd" value="0.00005625"/>
    </node>
    <node pkg="challenge_2" exec="set_point" name="sp_gen" namespace="gpo2">
      <param name="amplitud" value="2.0"/>
      <param name="omega" value="1.0"/>
      <param name="timer_period" value="0.1"/>
      <param name="signal_type" value="3"/>
    </node>

    <node pkg="challenge_2" exec="dc_motor" name="motor_sys" namespace="gpo3">
      <param name="sample_time" value="0.01"/>
      <param name="K" value="2.16"/>
      <param name="tau" value="0.05"/>
      <param name="initial_conditions" value="0.0"/>
    </node>
    <node pkg="challenge_2" exec="controller" name="ctrl" namespace="gpo3">
      <param name="Kp" value="0.02255"/>
      <param name="Ki" value="1.1563"/>
      <param name="Kd" value="0.00005625"/>
    </node>
      
    <node pkg="challenge_2" exec="set_point" name="sp_gen" namespace="gpo3">
      <param name="amplitud" value="2.0"/>
      <param name="omega" value="1.0"/>
      <param name="timer_period" value="0.1"/>
      <param name="signal_type" value="3"/>
    </node>


    <node pkg="rqt_plot" exec="rqt_plot" args='gpo1/set_point/data gpo1/motor_input_u/data gpo1/motor_output_y/data'/>
    <node pkg="rqt_graph" exec="rqt_graph"/>

</launch>
